packages/flutter/lib/src/rendering/object.dart

```diff
@@ -605,10 +605,10 @@ class _SemanticsGeometry {
     semantics.transform = _transform;
     if (_clipRect != null) {
       semantics.rect = _clipRect.intersect(rendering.semanticBounds);
-      semantics.wasAffectedByClip = true;
+      semantics.wasAffectedByClip = semantics.rect != rendering.semanticBounds;
     } else {
       semantics.rect = rendering.semanticBounds;
-      semantics.wasAffectedByClip = parentSemantics?.wasAffectedByClip ?? false;
+      semantics.wasAffectedByClip = false;
```

Comment:
wasAffectedByClip is defined as being "Whether rect might have been influenced by clips applied by ancestors", so why is this wrong?

Reply:
I am slightly redefining the meaning of wasAffectedByClip as indicated by the new doc comment and the description of this PR. It's only set to true if the rect of the SemanticsNode has actually changed due to clipping. I find that information more useful. Previously, when there was one clipped node in the semantics tree all of its descendants would be marked as clipped. That made it impossible to figure out which children actually had their size changed due to clipping.

Reply:
That's totally fine but then you should update the documentation to match :-)

Reply:
Ohh, I did: https://github.com/flutter/flutter/pull/12358/files#diff-647e4aa016f360586e38d3cdf5bad6f0R288

I will make that even clearer by adding some more information from my previous comment in this thread.